addons/website_event_track/tests/test_website.py

```diff
@@ -19,3 +23,48 @@ def test_event_app_name(self):
         self.assertEqual(website2.events_app_name, 'Foo Events')
         website2.write({'name': 'Bar'})
         self.assertEqual(website2.events_app_name, 'Foo Events')
+
+    def test_compute_app_icon(self):
+        # Load sample image files for testing
+        png_file_path = get_module_resource('website_event_track', 'tests', 'test.png')
+        jpeg_file_path = get_module_resource('website_event_track', 'tests', 'test.jpeg')
+        svg_file_path = get_module_resource('website_event_track', 'tests', 'test.svg')
+        jpg_file_path = get_module_resource('website_event_track', 'tests', 'test.jpg')
+
+        # Image file paths and their respective expected types
+        image_paths = {
+            png_file_path: 'png',
+            jpeg_file_path: 'jpeg',
+            jpg_file_path: 'jpg',
+            svg_file_path: 'svg'
+        }
+
+        for image_path, expected_type in image_paths.items():
+            # Read the image file
+            with open(image_path, 'rb') as image_file:
+                image_data = image_file.read()
+
+            # Create a website record
+            website = self.env['website'].create({
+                'name': 'Test Website',
+                'favicon': base64.b64encode(image_data)
+            })
+
+            # Call the method to compute app_icon
+            website._compute_app_icon()
+
+            if expected_type in ['jpeg', 'png', 'jpg']:
+                # Check if app_icon is set
+                self.assertTrue(website.app_icon)
+
+                # Check if app_icon is a valid image
+                decoded_image = base64.b64decode(website.app_icon)
+                image_io = BytesIO(decoded_image)
+                image = Image.open(image_io)
+                self.assertEqual(image.format.lower(), 'png')
+            else:
+                # For SVG images, ensure that the app_icon is not set
+                self.assertFalse(website.app_icon)
+
+            # Clean up the test data
+            website.unlink()
```

Comment:
Tests have a special cursor that handles resetting the database after each test, no need to unlink it yourself :)